src/Compilers/CSharp/Portable/Symbols/Wrappers/WrapperFieldSymbol.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a field that is based on another field.
+    /// When inheriting from this class, one shouldn't assume that 
+    /// the default behavior it has is appropriate for every case.
+    /// That behavior should be carefully reviewed and derived type
+    /// should override behavior as appropriate.
+    /// </summary>
+    internal abstract class WrapperFieldSymbol : FieldSymbol
```

Comment:
It would be more consistent with our existing naming conventions to call this WrappedFieldSymbol (and the same for all the others).